src/ComponentRegister.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+import {
+  ComponentRegister,
+  LinkComponent,
+  LinkComponentProps
+} from "./ComponentRegister";
+
+const TestLinkComponent: LinkComponent = ({
+  href,
+  children
+}: LinkComponentProps) => <div data-href={href}>{children}</div>;
+
+describe(".init()", () => {
+  it("registers all of the components provided", () => {
+    ComponentRegister.init({
+      components: {
+        Link: TestLinkComponent
+      }
+    });
+
+    expect(ComponentRegister.Link).toStrictEqual(TestLinkComponent);
+  });
+  it("does not alter properties if none are provided", () => {
+    ComponentRegister.init({});
+
+    expect(ComponentRegister.Link).toStrictEqual("a");
```

Comment:
```suggestion
    expect(ComponentRegister.Link).toEqual("a");
```

It's not a big deal, but in this case, we don't care about strict equality. It doesn't need to be the exact instance of `"a"` provided. For strings, this doesn't matter, but for components, this would probably be a false negative.